Update GetRecords in AggregationProcess to return map format records

yanjunz97 opened the pull request:
This commit updates the return type of GetRecords in AggregationProcess from string to map, which makes it more easier to be structured.

Signed-off-by: Yanjun Zhou <zhouya@vmware.com>

yanjunz97 commented:
As Antonin suggests, the record should be able to be encoded to JSON format. I changed the return type of the `GetRecords` function to a map of interface. Currently the `GetString()` function in Record is not used. Shall we still keep it?

srikartati commented:
> As Antonin suggests, the record should be able to be encoded to JSON format. I changed the return type of the `GetRecords` function to a map of interface. Currently the `GetString()` function in Record is not used. Shall we still keep it?

I think JSON format is good right. We can leave it to the user to change in the format they want. If we need some other format, we can add it later. Lets remove for now?

yanjunz97 commented:
> > As Antonin suggests, the record should be able to be encoded to JSON format. I changed the return type of the `GetRecords` function to a map of interface. Currently the `GetString()` function in Record is not used. Shall we still keep it?
> 
> I think JSON format is good right. We can leave it to the user to change in the format they want. If we need some other format, we can add it later. Lets remove for now?

Removed.

srikartati commented on pkg/intermediate/aggregate.go:
```diff
@@ -215,19 +215,19 @@ func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
-// Returns partial match if FlowKey is not complete.
-// Returns all the flow records if FlowKey is not provided.
-func (a *AggregationProcess) GetRecords(flowKey *FlowKey) []string {
+// GetRecords returns map format flow records given a flow key.
```
Let's explain the map format clearly, where key is element name and value is the IE object.

yanjunz97 replied:
Done

srikartati commented on pkg/intermediate/aggregate.go:
```diff
@@ -215,19 +215,19 @@ func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
-// Returns all the flow records if FlowKey is not provided.
-func (a *AggregationProcess) GetRecords(flowKey *FlowKey) []string {
+// GetRecords returns map format flow records given a flow key.
+// Returns partial matched flow records if the flow key is not complete.
```
"partially matched"

yanjunz97 replied:
Done

srikartati reviewed (approved):
lgtm

dreamtalen reviewed (approved):
LGTM
